src/main/metabase/card-layout.ts

```diff
@@ -0,0 +1,90 @@
+export type Rectangle = {
+  row: number;
+  col: number;
+  sizeX: number;
+  sizeY: number;
+};
+
+const ROW_WIDTH = 18;
+
+function validateSize(sizeX: number, sizeY: number) {
+  if (sizeX <= 0 || sizeX > ROW_WIDTH || sizeY <= 0) {
+    throw new Error('Invalid rectangle size.');
+  }
+}
+
+export class CardLayout {
```

Comment:
maybe such logic-heavy piece of code could use a unit test?

Reply:
Yep, I've been thinking about it.